search.py

```diff
@@ -132,23 +156,43 @@ def find_term(term):
     Takes in a term, then finds and returns the list representation of the PostingList of the given term
     or an empty list if no such term exists in index
     """
-    term = term.strip()
+    # NOTE: LOWERCASING IS ONLY DONE HERE.
+    term = term.strip().lower()
     if term not in D:
-        return []
+        return None
     POSTINGS_FILE_POINTER.seek(D[term])
-    return decode(pickle.load(POSTINGS_FILE_POINTER).postings)
+    return pickle.load(POSTINGS_FILE_POINTER)
+
+def find_by_document_id(terms):
+    """
+    Checks if any of the query terms are document ids, if so return the document id
+    To be used after the normal boolean/free text parsing
+    """
+    document_ids = []
+    for term in terms:
+        if all(map(str.isdigit, term)):
+            if int(term) in ALL_DOC_IDS:
+                document_ids.append(int(term))
+    return document_ids
 
 # Takes in a phrasal query in the form of an array of terms and returns the doc ids which have the phrase
 # Note: Only use this for boolean retrieval, not free text mode
-def perform_phrase_query(phrase):
+def perform_phrase_query(phrase_query):
     # Defensive programming, if phrase is empty, return false
-    if not phrase:
+    if not phrase_query:
         return False
-    phrase_posting_list = find_term(phrase[0])
-    for term in phrase[1:]:
+    phrases = phrase_query.split(" ")
+    phrase_posting_list = find_term(phrases[0].strip())
```

Comment:
is it haven't remove the " in the terms of phrasal query yet?

Reply:
cause the filter_punctuation doesn't remove "

Reply:
This one splits by space though

Reply:
Somehow the positional indices are off. There are documents with consecutive terms but somehow they dont have consecutive positions